### PR TITLE
Prevent sv_alltalk 1 exposing cloaked spies who are talking

### PR DESCRIPTION
**Pull request by Thestickman391**

### CODE REVIEW

**Comment by micwoj92** [comment]
This should be made into cvar and old behaviour set as default.

**Comment by Yttrium-tYcLief** [comment]
Why would we *want* this to persist in the game?

**Review by mastercoms — APPROVED** [review]

**Review by mastercoms — CHANGES_REQUESTED** [review]
Could you check for sv_alltalk?

**Comment by micwoj92** [comment]
> Why would we _want_ this to persist in the game?

Trolling.

File: game/client/tf/vgui/tf_overview.cpp [hunk 1]
```diff
@@ -503,7 +503,7 @@ void CTFMapOverview::DrawVoiceIconForPlayer( int playerIndex )
 	// if they just sent a chat msg, or are using voice, or did a hand signal or voice command
 	// draw a chat icon
 
-	if ( cl_voicetest.GetInt() || GetClientVoiceMgr()->IsPlayerSpeaking( player->index+1 ) )
+	if ( cl_voicetest.GetInt() || GetClientVoiceMgr()->IsPlayerSpeaking( player->index+1 ) && !pPlayer->m_Shared.IsFullyInvisible() || InSameTeam( pPlayer ) )
```
Comment (wgetJane) [inline game/client/tf/vgui/tf_overview.cpp:506]:
```suggestion
	if ( cl_voicetest.GetInt() || ( GetClientVoiceMgr()->IsPlayerSpeaking( player->index+1 ) && ( !pPlayer->m_Shared.IsFullyInvisible() || InSameTeam( pPlayer ) ) ) )
```

fix this logical expression

**Review by wgetJane — CHANGES_REQUESTED** [review]
this looks it would cause the speech bubble to always appear for teammates even if they're not speaking

**Comment by maximsmol** [comment]
Besides adding a check for `sv_alltalk` we should also actually test it to make sure @wgetJane's version of the check works properly (I am rather certain the old one was broken)

**Comment by Thestickman391** [comment]
Added a check for sv_alltalk

**Comment by maximsmol** [comment]
Ok, perfect. Assuming that this was tested and merging since Jane saw it too.

**Pull request merged**